### PR TITLE
Add some scripts and regions to HelloWorldProvider

### PR DESCRIPTION
**Pull request by sffc**
Thanks @nciric for the contribution!

(I need this for making an interesting example in #3928)

### CODE REVIEW

**Review by zbraniecki — APPROVED** [review]

**Comment by sffc** [comment]
```
warning: spurious network error (3 tries remaining): [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
warning: spurious network error (2 tries remaining): [28] Timeout was reached (Operation too slow. Less than [10](https://github.com/unicode-org/icu4x/actions/runs/5969184284/job/16194546516?pr=3936#step:8:11) bytes/sec transferred the last 30 seconds)
warning: spurious network error (1 tries remaining): [28] Timeout was reached (Operation too slow. Less than 10 bytes/sec transferred the last 30 seconds)
error: failed to compile `cargo-all-features v1.10.0`, intermediate artifacts can be found at `/tmp/cargo-installeT1DY4`

Caused by:
  failed to get `syn` as a dependency of package `clap_derive v4.4.0`
```

**Comment by robertbastian** [comment]
What can you demonstrate with serbian that you cannot demonstrate with one of the existing languages?

**Comment by sffc** [comment]
> What can you demonstrate with serbian that you cannot demonstrate with one of the existing languages?

Only that it is more interesting when a script is involved in the fallback.

**Review by Manishearth — APPROVED** [review]

File: provider/core/src/hello_world.rs [hunk 1]
```diff
@@ -108,6 +108,7 @@ impl HelloWorldProvider {
         ("el", "Καλημέρα κόσμε"),
         ("en", "Hello World"),
         ("en+reverse", "Olleh Dlrow"),
+        ("en-GB", "’allo Mate!"),
```
Comment (robertbastian) [inline provider/core/src/hello_world.rs:111]: that's ... borderline offensive

Comment (sffc) [reply provider/core/src/hello_world.rs:111]: It's intended as a cheesy string for demonstrating regional variants.

Comment (sffc) [reply provider/core/src/hello_world.rs]: Changed to flag emoji

**Comment by sffc** [comment]
> I don't like this

I'm making this PR primarily in response to feedback that we should be using HelloWorldProvider instead of `icu_testdata` in tutorials and tests that don't work with compiled data. In order to exercise more code paths, it is useful to have multiple scripts and regions in the hello world data.

Also, I kind-of like how this works even better than `icu_testdata` for those use cases. It's easy to tell the locale that the data corresponds to.

**Review by robertbastian — APPROVED** [review]

**Pull request merged**